lib/http/timeout/null.rb

```diff
@@ -36,8 +36,9 @@ def start_tls(host, ssl_socket_class, ssl_context)
         connect_ssl
 
         return unless ssl_context.verify_mode == OpenSSL::SSL::VERIFY_PEER
+        return unless ssl_context.respond_to?(:verify_hostname) && ssl_context.verify_hostname
```

Comment:
In hindsight, this should probably have been:

```ruby
return if ssl_context.respond_to?(:verify_hostname) && !ssl_context.verify_hostname
```

...so as to only support explicitly *disabling* hostname verification. As is, it seems hostname verification won't be performed on JRuby at all.

Reply:
Opened #636 as a follow-up to this comment.